tests/paddle/test_paddle_model_export.py

```diff
@@ -295,3 +295,168 @@ def test_model_log_without_specified_conda_env_uses_default_env_with_expected_de
+
+    low_level_test_dataset = [x[0] for x in test_dataset]
+
+    np.testing.assert_array_almost_equal(
+        np.array(model.predict(test_dataset)).squeeze(),
+        np.array(reloaded_model(np.array(low_level_test_dataset))).squeeze(),
+    )
+
+
+@pytest.mark.large
+def test_model_built_in_high_level_api_log(pd_model_built_in_high_level_api, model_path):
+    old_uri = mlflow.get_tracking_uri()
+    model = pd_model_built_in_high_level_api.model
+    test_dataset = pd_model_built_in_high_level_api.inference_dataframe
+    with TempDir(chdr=True, remove_on_exit=True) as tmp:
+        for should_start_run in [False, True]:
+            try:
+                mlflow.set_tracking_uri("test")
+                if should_start_run:
+                    mlflow.start_run()
+
+                artifact_path = "model"
+                conda_env = os.path.join(tmp.path(), "conda_env.yaml")
+                _mlflow_conda_env(conda_env, additional_pip_deps=["paddle"])
+
+                mlflow.paddle.log_model(
+                    pd_model=model, artifact_path=artifact_path, conda_env=conda_env
+                )
+                model_uri = "runs:/{run_id}/{artifact_path}".format(
+                    run_id=mlflow.active_run().info.run_id, artifact_path=artifact_path
+                )
+
+                reloaded_pd_model = mlflow.paddle.load_model(model_uri=model_uri)
+
+                low_level_test_dataset = [x[0] for x in test_dataset]
+
+                np.testing.assert_array_almost_equal(
+                    np.array(model.predict(test_dataset)).squeeze(),
+                    np.array(reloaded_pd_model(np.array(low_level_test_dataset))).squeeze(),
+                )
+
+                model_path = _download_artifact_from_uri(artifact_uri=model_uri)
+                model_config = Model.load(os.path.join(model_path, "MLmodel"))
+                assert pyfunc.FLAVOR_NAME in model_config.flavors
+                assert pyfunc.ENV in model_config.flavors[pyfunc.FLAVOR_NAME]
+                env_path = model_config.flavors[pyfunc.FLAVOR_NAME][pyfunc.ENV]
+                assert os.path.exists(os.path.join(model_path, env_path))
+
+            finally:
+                mlflow.end_run()
+                mlflow.set_tracking_uri(old_uri)
+
+
+@pytest.fixture
+def model_retrain_path(tmpdir):
+    return os.path.join(str(tmpdir), "model_retrain")
+
+
+@pytest.mark.large
+def test_model_retrain_built_in_high_level_api(
+    pd_model_built_in_high_level_api, model_path, model_retrain_path
+):
+    model = pd_model_built_in_high_level_api.model
+    mlflow.paddle.save_model(pd_model=model, path=model_path, training=True)
+
+    class UCIHousing(paddle.nn.Layer):
+        def __init__(self):
+            super(UCIHousing, self).__init__()
+            self.fc_ = paddle.nn.Linear(13, 1, None)
+
+        def forward(self, inputs):  # pylint: disable=arguments-differ
+            pred = self.fc_(inputs)
+            return pred
```

Comment:
Can we define this class in the __main__ scope?

Reply:
Sure! The code is much cleaner now.